server/book/book.go

```diff
@@ -26,41 +25,27 @@ const (
 type Book struct {
 	mtx     sync.RWMutex
 	lotSize uint64
-	halfCap uint32
 	buys    *OrderPQ
 	sells   *OrderPQ
 }
 
 // New creates a new order book with the given lot size, and optional order
 // capacity of each side of the book. To change capacity of an existing Book,
 // use the Realloc method.
-func New(lotSize uint64, halfCapacity ...uint32) *Book {
-	halfCap := DefaultBookHalfCapacity
-	if len(halfCapacity) > 0 {
-		halfCap = halfCapacity[0]
-	}
+func New(lotSize uint64) *Book {
 	return &Book{
 		lotSize: lotSize,
-		halfCap: halfCap,
-		buys:    NewMaxOrderPQ(halfCap),
-		sells:   NewMinOrderPQ(halfCap),
+		buys:    NewMaxOrderPQ(initBookHalfCapacity),
+		sells:   NewMinOrderPQ(initBookHalfCapacity),
 	}
 }
 
 // Clear reset the order book with configured capacity.
 func (b *Book) Clear() {
 	b.mtx.Lock()
-	b.buys = NewMaxOrderPQ(b.halfCap)
-	b.sells = NewMaxOrderPQ(b.halfCap)
-	b.mtx.Unlock()
-}
-
-// Realloc changes the capacity of the order book given the specified capacity
-// of both buy and sell sides of the book.
-func (b *Book) Realloc(newHalfCap uint32) {
-	b.mtx.Lock()
-	b.buys.Realloc(newHalfCap)
-	b.sells.Realloc(newHalfCap)
+	b.buys, b.sells = nil, nil
```

Comment:
Yeah, but maybe it's fruitless.  In a memory constrained environment and large capacity queues, I encountered an OOM in Clear so I'm letting the GC know that it can eliminate the old ones if needed sooner than later.  Not sure if this is effective though... hence this PR.